src/codegen/mod.rs

```diff
@@ -1832,16 +1832,12 @@ impl From<&ast::Builtin> for Builtin {
     }
 }
 
-pub(super) fn error_msg_with_loc(ns: &Namespace, error: &str, loc: Option<Loc>) -> String {
-    if let Some(loc) = loc {
-        match loc {
-            Loc::File(..) => {
-                let loc_from_file = ns.loc_to_string(PathDisplay::Filename, &loc);
-                format!("runtime_error: {error} in {loc_from_file}")
-            }
-            _ => error.to_string(),
+pub(super) fn error_msg_with_loc(ns: &Namespace, error: String, loc: Option<Loc>) -> String {
+    match &loc {
+        Some(loc @ Loc::File(..)) => {
+            let loc_from_file = ns.loc_to_string(PathDisplay::Filename, loc);
+            format!("runtime_error: {error} in {loc_from_file},\n")
         }
-    } else {
-        error.to_string()
+        _ => error + ",\n",
```

Comment:
I know you did not introduce this, but why the trailing `,`?

Reply:
I guess that @salaheldinsoliman's reasoning was that there can be multiple errors from a file, so he would split them using a comma.

Reply:
As far as I remember the contracts node would print (entered unreachable code) or sth like that after the error is printed, so that makes the error more readable

Reply:
On Substrate, the strings get concatenated to one big string, while on Solana you get a vector of strings. Maybe the `,` makes more sense on Substrate that Solana. We can leave it for now